File: src/services/system.ts
import { invoke } from '@tauri-apps/api/core';
import type { PlatformInfo, DiagnosticsInfo } from '../shared/types';

export async function getPlatformInfo(): Promise<PlatformInfo> {
  return await invoke('get_platform_info');
}

export async function getDiagnosticsInfo(): Promise<DiagnosticsInfo> {
  return await invoke('get_diagnostics_info');
}

File: src-tauri/src/commands/system.rs
use crate::error::AppError;
use crate::platform::models::{DiagnosticsInfo, PlatformInfo};
use crate::state::AppState;
use tauri::State;

/// Get detailed platform information.
#[tauri::command]
pub fn get_platform_info(state: State<'_, AppState>) -> Result<PlatformInfo, AppError> {
    Ok(state.platform.info().clone())
}

/// Get runtime health diagnostics.
#[tauri::command]
pub fn get_diagnostics_info(state: State<'_, AppState>) -> Result<DiagnosticsInfo, AppError> {
    Ok(state.platform.diagnostics())
}
